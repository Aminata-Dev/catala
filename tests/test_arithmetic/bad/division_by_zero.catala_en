## `Division_by_zero` exception management

### with integers

```catala
declaration scope Int:
  context i content decimal

scope Int:
  definition i equals 1 / 0
```

### with decimals

```catala
declaration scope Dec:
  context i content decimal

scope Dec:
  definition i equals 1. / 0.
```

### with money

```catala
declaration scope Money:
  context i content decimal

scope Money:
  definition i equals $10.0 / $0.0
```


```catala-test-inline
$ catala Interpret -s Dec
[WARNING] In scope "Int", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:7.10-11:
└─┐
7 │   context i content decimal
  │           ‾
  └┬ `Division_by_zero` exception management
   └─ with integers
[WARNING] In scope "Dec", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:17.10-11:
└──┐
17 │   context i content decimal
   │           ‾
   └┬ `Division_by_zero` exception management
    └─ with decimals
[WARNING] In scope "Money", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:27.10-11:
└──┐
27 │   context i content decimal
   │           ‾
   └┬ `Division_by_zero` exception management
    └─ with money
[ERROR] division by zero at runtime

The division operator:
┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:20.22-29:
└──┐
20 │   definition i equals 1. / 0.
   │                       ‾‾‾‾‾‾‾
   └┬ `Division_by_zero` exception management
    └─ with decimals

The null denominator:
┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:20.27-29:
└──┐
20 │   definition i equals 1. / 0.
   │                            ‾‾
   └┬ `Division_by_zero` exception management
    └─ with decimals
#return code 255#
```

```catala-test-inline
$ catala Interpret -s Int
[WARNING] In scope "Int", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:7.10-11:
└─┐
7 │   context i content decimal
  │           ‾
  └┬ `Division_by_zero` exception management
   └─ with integers
[WARNING] In scope "Dec", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:17.10-11:
└──┐
17 │   context i content decimal
   │           ‾
   └┬ `Division_by_zero` exception management
    └─ with decimals
[WARNING] In scope "Money", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:27.10-11:
└──┐
27 │   context i content decimal
   │           ‾
   └┬ `Division_by_zero` exception management
    └─ with money
[ERROR] division by zero at runtime

The division operator:
┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:10.22-27:
└──┐
10 │   definition i equals 1 / 0
   │                       ‾‾‾‾‾
   └┬ `Division_by_zero` exception management
    └─ with integers

The null denominator:
┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:10.26-27:
└──┐
10 │   definition i equals 1 / 0
   │                           ‾
   └┬ `Division_by_zero` exception management
    └─ with integers
#return code 255#
```

```catala-test-inline
$ catala Interpret -s Money
[WARNING] In scope "Int", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:7.10-11:
└─┐
7 │   context i content decimal
  │           ‾
  └┬ `Division_by_zero` exception management
   └─ with integers
[WARNING] In scope "Dec", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:17.10-11:
└──┐
17 │   context i content decimal
   │           ‾
   └┬ `Division_by_zero` exception management
    └─ with decimals
[WARNING] In scope "Money", the variable "i" is never used anywhere; maybe it's unnecessary?

┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:27.10-11:
└──┐
27 │   context i content decimal
   │           ‾
   └┬ `Division_by_zero` exception management
    └─ with money
[ERROR] division by zero at runtime

The division operator:
┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:30.22-34:
└──┐
30 │   definition i equals $10.0 / $0.0
   │                       ‾‾‾‾‾‾‾‾‾‾‾‾
   └┬ `Division_by_zero` exception management
    └─ with money

The null denominator:
┌─⯈ tests/test_arithmetic/bad/division_by_zero.catala_en:30.30-34:
└──┐
30 │   definition i equals $10.0 / $0.0
   │                               ‾‾‾‾
   └┬ `Division_by_zero` exception management
    └─ with money
#return code 255#
```
