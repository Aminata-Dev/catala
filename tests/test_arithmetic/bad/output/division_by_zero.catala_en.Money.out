[ERROR] division by zero at runtime

The division operator:
  --> test_arithmetic/bad/division_by_zero.catala_en
   | 
30 |   definition i equals $10.0 /$ $0.0
   |                             ^^
   + `Division_by_zero` exception management
   +-+ with money

The null denominator:
  --> test_arithmetic/bad/division_by_zero.catala_en
   | 
30 |   definition i equals $10.0 /$ $0.0
   |                                ^^^^
   + `Division_by_zero` exception management
   +-+ with money
